tasks/create_update_quadlet_spec.yml

```diff
@@ -0,0 +1,145 @@
+---
+- name: Check if user is lingering
+  stat:
+    path: "/var/lib/systemd/linger/{{ __podman_user }}"
+  register: __podman_user_lingering
+  when: __podman_rootless | bool
+
+- name: Enable lingering if needed
+  command: loginctl enable-linger {{ __podman_user }}
+  when:
+    - __podman_rootless | bool
+    - not __podman_user_lingering.stat.exists
+  changed_when: true
+
+- name: Create host directories
+  file: "{{ __defaults | combine(podman_host_directories[__hostitem])
+            if __hostitem in podman_host_directories | d({})
+            else __defaults }}"
+  vars:
+    __hostitem: "{{ item if item in podman_host_directories | d({})
+      else 'DEFAULT' }}"
+    __owner_group: "{{ {'owner': __podman_user, 'group': __podman_group}
+      if __podman_rootless else {'owner': 'root', 'group': 'root'} }}"
+    __defaults: "{{ {'path': item} | combine(__podman_hostdirs_defaults) |
+      combine(__owner_group) }}"
+  loop: "{{ __podman_volumes }}"
+  become: "{{ __podman_rootless | ternary(true, omit) }}"
+  become_user: "{{ __podman_rootless | ternary(__podman_user, omit) }}"
+  when:
+    - podman_create_host_directories | bool
+    - __podman_volumes | d([]) | length > 0
+
+- name: Ensure container images are present
+  containers.podman.podman_image:
+    name: "{{ item }}"
+    force: true
+    username: "{{ container_image_user | default(omit) }}"
+    password: "{{ container_image_password | default(omit) }}"
+  register: __podman_image_updated
+  environment:
+    XDG_RUNTIME_DIR: "{{ __podman_xdg_runtime_dir }}"
+  become: "{{ __podman_rootless | ternary(true, omit) }}"
+  become_user: "{{ __podman_rootless | ternary(__podman_user, omit) }}"
+  loop: "{{ __podman_images | unique | list }}"
+
+- name: Ensure the quadlet directory is present
+  file:
+    path: "{{ __podman_quadlet_file | dirname }}"
+    state: directory
+    owner: "{{ __podman_user }}"
+    group: "{{ __podman_group }}"
+    mode: "0755"
+  when: not __podman_quadlet_file is none
```

Comment:
I changed all of these.  I believe we had to use the `in [none, ""]` form with older versions of ansible and/or jinja.  I think we don't need to use this anymore.

Reply:
So, there is a bit of a wrinkle - if you use `when: __some_string_valued_variable`, you will get the "bare variable" warning - https://github.com/redhat-cop/automation-good-practices/tree/main/coding_style#ansible-guidelines - the recommendation is to use `when: __some_string_valued_variable | bool` - this works fine if the variable is empty or `none` - however, when it is a non-zero length string, `when: __some_string_valued_variable | bool` will evaluate to `false` because it is doing some sort of string-to-bool conversion - so you must use `when: __some_string_valued_variable | length > 0` to check for a non-empty string.